### PR TITLE
🐛 Bug Fix: Show Less Button Incorrect Rendering

### PR DESCRIPTION
**Pull request by SAHU-01**
Fix for #489 

Describe changes proposed in this pull request:

- Add a unique cell identifying value for state management, so that **_show more_** and **_show less_** act for that unique cell alone.

Screenshot:

Given:
image

Before:
image

After:
image

Notify reviewers:
@onursumer

### CODE REVIEW

File: packages/data-portal-commons/src/components/ExpandableText.tsx [hunk 1]
```diff
@@ -25,13 +26,17 @@ export const ExpandableText: React.FunctionComponent<ExpandableTextProps> = (
     props
 ) => {
     // truncate text by default
-    const [truncated, setTruncated] = useState<boolean>(true);
+    const [truncated, setTruncated] = useState<{ [key: string]: boolean }>({});
```
Comment (onursumer) [inline packages/data-portal-commons/src/components/ExpandableText.tsx:29]: It's confusing for this component to keep the state of multiple values when it only receives a single text value.

File: packages/data-portal-commons/src/components/ExpandableText.tsx [hunk 2]
```diff
@@ -25,13 +26,17 @@ export const ExpandableText: React.FunctionComponent<ExpandableTextProps> = (
...
 
     const showMoreText = props.showMoreText || 'show more';
     const showLessText = props.showLessText || 'show less';
+    const cellKey = `${props.value}`;
```
Comment (onursumer) [inline packages/data-portal-commons/src/components/ExpandableText.tsx:33]: Not sure if we should introduce the cell key inside this component. What is a cell in the context of this component?

File: packages/data-portal-explore/src/lib/dataTableHelpers.tsx [hunk 3]
```diff
@@ -147,8 +147,7 @@ export function truncatedTableCell<T>(cellData: T) {
     const value = _.isFunction(selector)
         ? selector(cellData)
         : cellData[selector as keyof T];
-
     return value ? (
-        <ExpandableText fullText={value} truncateProps={{ lines: 4 }} />
+        <ExpandableText fullText={value} truncateProps={{ lines: 4 }} value={value}/>
```
Comment (onursumer) [inline packages/data-portal-explore/src/lib/dataTableHelpers.tsx:151]: It's a bit confusing to pass the same value as two different props. We should probably come up with a better solution.

File: packages/data-portal-commons/src/components/ExpandableText.tsx (hunk 2 above)
Comment (SAHU-01) [reply packages/data-portal-commons/src/components/ExpandableText.tsx:33]: cellKey in our component acts as a unique identifier for each row's text value within the component's state. Since each row's text can be expanded or collapsed independently, the component needs a way to track the expand/collapse state for each row separately.

File: packages/data-portal-commons/src/components/ExpandableText.tsx [hunk 4]
```diff
@@ -40,6 +40,11 @@ export const ExpandableText: React.FunctionComponent<ExpandableTextProps> = (
         </span>
     );
 
+    useEffect(() => {
+            setTruncated(true);
+          }, [props.fullText]);
```
Comment (onursumer) [inline packages/data-portal-commons/src/components/ExpandableText.tsx:45]: not sure why but the indentation looks incorrect here, can you make sure that the `prettier` hook is running properly?

File: packages/data-portal-explore/src/lib/dataTableHelpers.tsx [hunk 5]
```diff
@@ -147,8 +147,7 @@ export function truncatedTableCell<T>(cellData: T) {
     const value = _.isFunction(selector)
         ? selector(cellData)
         : cellData[selector as keyof T];
-
     return value ? (
-        <ExpandableText fullText={value} truncateProps={{ lines: 4 }} />
+        <ExpandableText fullText={value} truncateProps={{ lines: 4 }}/>
```
Comment (onursumer) [inline packages/data-portal-explore/src/lib/dataTableHelpers.tsx:151]: we should just revert the changes in this file for simplicity. we don't really change anything but whitespaces.

**Review by onursumer — CHANGES_REQUESTED** [review]
Thanks for fixing this!

Just requested a few minor changes related to code formatting. Other than that it looks good :raised_hands:

File: packages/data-portal-explore/src/lib/dataTableHelpers.tsx [hunk 6]
```diff
@@ -147,7 +147,7 @@ export function truncatedTableCell<T>(cellData: T) {
     const value = _.isFunction(selector)
         ? selector(cellData)
         : cellData[selector as keyof T];
-
+
```
Comment (onursumer) [inline packages/data-portal-explore/src/lib/dataTableHelpers.tsx:150]: Looks like this file is not completely reverted. You can probably use some git revert / reset command.

**Review by onursumer — APPROVED** [review]
